internal/utils/diff.go

```diff
@@ -42,8 +42,32 @@ func GetRemoteDiffForConditions(source, remote []remoteconfig.Condition) string
 
 func GetRemoteDiffForParameters(source, remote map[string]remoteconfig.Parameter) string {
 	diff := cmp.Diff(remote, source)
-	greenDiff := strings.ReplaceAll(diff, "\n+", "\n"+Green)
+	maskedString := MaskSecrets(diff)
+	greenDiff := strings.ReplaceAll(maskedString, "\n+", "\n"+Green)
 	redDiff := strings.ReplaceAll(greenDiff, "\n-", "\n"+Red)
 	finalDiff := strings.ReplaceAll(redDiff, "\n", Reset+"\n")
 	return finalDiff
 }
+
+func MaskSecrets(input string) string {
```

Comment:
why is this an exported function its only used within the same package right ?

Reply:
@adarshk-rapido yes, this is being used within the same package. This need not be exported. I have made the change.